src/App.js

```diff
@@ -1,23 +1,133 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+
+import Header from "./components/Header";
+import DisplayAlgo from "./components/DisplayAlgo";
+import DisplayAlgoLine from "./components/DisplayAlgoLine";
+import Footer from "./components/Footer";
+import { useState, useEffect } from "react";
 
 function App() {
+  const numRow = Math.floor(window.innerWidth / 11);
+  const [runningAlgorithm, setRunningAlgorithm] = useState(false);
+  const [rows, setRows] = useState([]);
+  const [arrayAccessCounter, setArrayAccessCounter] = useState(0);
+  const [comparisonCounter, setComparisonCounter] = useState(0);
+  //_____________________Functions________________________________________
+
+  //randomize graph
+  const randomize = () => {
+    const tempArr = [];
+    for (var i = 0; i < numRow; i++) {
+      tempArr.push({
+        height: Math.floor(Math.random() * 500),
+        id: i,
+      });
+    }
+    setRows((rows) => tempArr);
```

Comment:
When using a state setter, if you don't need to access the current value, you can probably rely on the short version as following:

```suggestion
    setRows(tempArr);
```